sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

```diff
@@ -19,41 +22,101 @@ public class TestRecording : IDisposable
+                        if (File.Exists(_sessionFile))
+                        {
+                            try
+                            {
+                                _previousSession = Load();
+                            }
+                            catch (Exception)
+                            {
+                                // ignore
+                            }
+                        }
+
+                        break;
+                    case RecordedTestMode.Playback:
                         try
                         {
-                            _previousSession = Load();
+                            Session = Load();
                         }
-                        catch (Exception)
+                        catch (Exception ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
                         {
-                            // ignore
+                            _mismatchException = new TestRecordingMismatchException(ex.Message, ex);
                         }
-                    }
-                    break;
-                case RecordedTestMode.Playback:
-                    try
-                    {
-                        Session = Load();
-                    }
-                    catch (Exception ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
-                    {
-                        _mismatchException = new TestRecordingMismatchException(ex.Message, ex);
-                    }
-                    break;
+
+                        break;
+                }
+            }
+            else
+            {
+                switch (Mode)
+                {
+                    case RecordedTestMode.Record:
+                        var recordResponse = _proxy.Client.StartRecord(_sessionFile);
+                        RecordingId = recordResponse.Headers.XRecordingId;
+                        AddProxySanitizers(_matcher.CompareBodies);
+
+                        break;
+                    case RecordedTestMode.Playback:
+                        ResponseWithHeaders<IReadOnlyDictionary<string, string>, TestProxyStartPlaybackHeaders> playbackResponse = null;
+                        try
+                        {
+                            playbackResponse = _proxy.Client.StartPlayback(_sessionFile);
+                        }
+                        catch (RequestFailedException ex)
+                            when (ex.Status == 500)
+                        {
+                            _mismatchException = new TestRecordingMismatchException(ex.Message, ex);
+                            break;
+                        }
+
+                        _variables = new SortedDictionary<string, string>((Dictionary<string, string>)playbackResponse.Value);
+                        RecordingId = playbackResponse.Headers.XRecordingId;
+                        AddProxySanitizers(_matcher.CompareBodies);
```

Comment:
Why are we adding sanitizers for playback?

Reply:
There are some tests that are loading non-sanitized variables from the recordings, but the variables get sanitized as part of the body in the request.